### PR TITLE
plat-stm32mp1: prepare for clock framework

### PR DESCRIPTION
**Pull request by etienne-lms**
A series of changes around resources to prepare support for generic clock API.

### CODE REVIEW

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c [hunk 1]
```diff
@@ -1306,20 +1306,28 @@ static TEE_Result stm32mp1_clk_early_init(void)
...
-	} else {
+	assert(_fdt_reg_base_address(fdt, node) == RCC_BASE);
+	assert(_fdt_get_status(fdt, node) != DT_STATUS_DISABLED);
+
+	if (!rcc_secure) {
 		if (io_read32(stm32_rcc_base() + RCC_TZCR) & RCC_TZCR_TZEN)
 			panic("Refuse to release RCC[TZEN]");
 
-		IMSG("RCC is non-secure");
+		disable_rcc_tzen();
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c]: At this point in the patch series, this function doesn't exist.

**Review by jforissier — COMMENTED** [review]
One issue below, otherwise LGTM:
`Acked-by: Jerome Forissier <jerome@forissier.org>`

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c]: Oh! i must have missed something! My apologies.

**Comment by etienne-lms** [comment]
I've fixed the faulty commit. Each build/boot ok now. Thanks for the catch.

Updated and A-b tag applied.

**Pull request merged**